File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    s = list(s)
    word = ''
    
    count = result = 0
    
    for i in range(0, len(s)):
        if(count == 0):
            word = s[i]
            result += 1
        
        if(s[i] == word):
            count += 1
            
        elif(s[i] != word):
            count -= 1

    return result

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
  word = ''
  count = 0
  result = 0

  for (i = 0; i < s.length; i++) {
    if (count == 0) {
      word = s[i];
      result += 1;
    }

    if (s[i] == word) {
      count += 1;
    } else if (s[i] !== word) {
      count -= 1;
    }
  }

  return result;
}